src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -494,7 +494,8 @@ private enum AssignmentKind
         /// </summary>
         private bool ReportNullableAssignmentIfNecessary(BoundExpression value, TypeSymbolWithAnnotations targetType, TypeSymbolWithAnnotations valueType, bool useLegacyWarnings, AssignmentKind assignmentKind = AssignmentKind.Assignment, Symbol target = null)
         {
-            Debug.Assert(!IsConditionalState);
+            // PROTOTYPE(NullableReferenceTypes): Need to discuss
+            //Debug.Assert(!IsConditionalState);
```

Comment:
This assert can be removed. Reporting warnings does not depend on whether the state is conditional.